annotations/views/repository_views.py

```diff
@@ -73,12 +73,21 @@ def retrieve(self, request, pk=None, repository_pk=None):
         try:
             master_text = Text.objects.get(uri=result.get('uri'))
         except Text.DoesNotExist:
-            master_text = None
+            master_text = Text.objects.create(uri=result.get('uri'),title=result.get('name'),public=result.get('public'),content_type=result.get('content_types'),repository_source_id=result.get('id'),repository_id=repository_pk,addedBy_id=1)
         aggregate_content = result.get('aggregate_content')
+
+        submitted = False
+        i = 0
+        while i < len(master_text.children):
```

Comment:
This is not a great way to do this. One it is very inefficient and seems to have created long load times. Two its not as readable as it could be and has added unneeded lines of code. Use a for loop to iterate over children and then add a break statement after submitted is changed to true. There is no point continue to iterate over the annotations once the condition is meet, it just wastes time and resources.